caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -0,0 +1,142 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Standard
+import os
+
+# Third Party
+from sentence_transformers import SentenceTransformer
+
+# First Party
+from caikit.core import ModuleBase, ModuleConfig, ModuleSaver, module
+from caikit.core.exceptions import error_handler
+import alog
+
+# Local
+from .embedding_tasks import EmbeddingTask
+from caikit_nlp.data_model.embedding_vectors import EmbeddingResult, Vector1D
+
+logger = alog.use_channel("TXT_EMB")
+error = error_handler.get(logger)
+
+
+@module(
+    "EEB12558-B4FA-4F34-A9FD-3F5890E9CD3F",
+    "EmbeddingModule",
+    "0.0.1",
+    EmbeddingTask,
+)
+class EmbeddingModule(ModuleBase):
+
+    _ARTIFACTS_PATH_KEY = "artifacts_path"
+    _ARTIFACTS_PATH_DEFAULT = "artifacts"
+
+    def __init__(
+        self,
+        model: SentenceTransformer,
+    ):
+        super().__init__()
+        self.model = model
+
+    @classmethod
+    def load(cls, model_path: str, *args, **kwargs) -> "EmbeddingModule":
+        """Load model
+
+        Args:
+            model_path: str
+                Path to the config dir under the model_id (where the config.yml lives)
+
+        Returns:
+            EmbeddingModule
+                Instance of this class built from the model.
+        """
+
+        config = ModuleConfig.load(model_path)
+        artifacts_path = config.get(cls._ARTIFACTS_PATH_KEY)
+
+        error.value_check(
+            "<NLP07391618E>",
+            artifacts_path,
+            ValueError(f"Model config missing '{cls._ARTIFACTS_PATH_KEY}'"),
+        )
+
+        artifacts_path = os.path.abspath(os.path.join(model_path, artifacts_path))
+        error.dir_check("<NLP34197772E>", artifacts_path)
+
+        return cls.bootstrap(model_name_or_path=artifacts_path)
+
+    def run(
+        self, input: str, **kwargs  # pylint: disable=redefined-builtin
+    ) -> EmbeddingResult:
+        """Run inference on model.
+        Args:
+            input: str
+                Input text to be processed
+        Returns:
+            EmbeddingResult: the result vector nicely wrapped up
+        """
+        error.type_check("<NLP27491611E>", str, input=input)
+
+        return EmbeddingResult(Vector1D.from_vector(self.model.encode(input)))
```

Comment:
do we need to do some validation on `model.encode` output before converting it to embedding via vector or is it guaranteed to be a compatible vector always?

Reply:
Always returns a tensor that works with Vector1D.from_vector().   There is nothing to validate here.  If the dtype is something we didn't expect we use PyFloatSequence().   I could add a __post_init__ so PyFloatSequence() evaluates each value, but I don't have a real case for that right now.